sphinxgallery/gen_rst.py

```diff
@@ -373,7 +387,7 @@ def execute_script(code_block, example_globals, image_path, fig_count,
     cwd = os.getcwd()
     # Redirect output to stdout and
     orig_stdout = sys.stdout
-
+    figure_list = []
```

Comment:
> What I see, is that this avoids a NameError if the code block did not execute correctly, leaving you out of the try-catch without the variable figure_list that you later need in the return statement.

You are right, it was actually only broken in case the example script had some errors.